### PR TITLE
SAM-1732 SAM-1733 Pulling messages and sync with postgres + Draining

### PR DESCRIPTION
**Pull request by chenrozenes**

### CODE REVIEW

File: src/lib/puller/postgresPuller.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+import logger from '../logger';
+import Puller from './puller';
+
+const TABLE_NAME = 'rabbit_pull_queue_sync';
+const CREATE_QUERY = `CREATE TABLE IF NOT EXISTS ${TABLE_NAME} (` +
+  'queue CHARACTER VARYING(200) NOT NULL UNIQUE, ' +
+  'last_pull TIMESTAMP NOT NULL);';
+
+const INIT_FIRST_VALUE = `INSERT INTO ${TABLE_NAME} VALUES($1,'-infinity') ON CONFLICT DO NOTHING`;
```
Comment (asafdav) [inline src/lib/puller/postgresPuller.js]: I think we should keep some statistics in each line (how many messages were sent in the last sec, minute, hours) so moving forward we'd be able to rate limit based on a more sophisticated approach

File: src/lib/puller/postgresPuller.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  /**
+   * Trying to update the queue row, if succeeded current worker is allowed to pull.
+   *
+   * @returns {*}
+   */
+  shouldPull() {
+    const now = moment();
+    const lessThen = moment(now).subtract(this.interval / 1000, 'seconds'); // Subtract interval
```
Comment (asafdav) [inline src/lib/puller/postgresPuller.js]: maybe move to a const

File: src/lib/puller/postgresPuller.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    return Promise.resolve()
+      .then(() => this.client.query(UPDATE_QUERY, [
```
Comment (asafdav) [inline src/lib/puller/postgresPuller.js]: If I understand correctly - we'll support only serial sending, we'd need to do more than that - like support max of 5 messages at once or X messages in a period of time

File: src/lib/puller/postgresPuller.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      .then(() => super.start(channel, queue, callback));
+  }
+
+  stop() {
+    return super.stop()
+      .then(() => this._disconnect());
+  }
+}
+
+export default PostgresPuller;
```
Comment (asafdav) [inline src/lib/puller/postgresPuller.js]: I like this idea of the puller - good job

File: src/lib/puller/postgresPuller.js (hunk 3 above)
Comment (chenrozenes) [reply src/lib/puller/postgresPuller.js]: I agree, but I prefer to move on now, do you think its urgent?

File: src/lib/puller/postgresPuller.js (hunk 2 above)
Comment (chenrozenes) [reply src/lib/puller/postgresPuller.js]: So I skipped that

File: src/lib/puller/postgresPuller.js (hunk 4 above)
Comment (chenrozenes) [reply src/lib/puller/postgresPuller.js]: @hartuvofir did a lot as well

File: src/lib/puller/postgresPuller.js (hunk 3 above)
Comment (chenrozenes) [reply src/lib/puller/postgresPuller.js]: I agree but its not trivial.
As we talked, I leave it that way for now

File: src/lib/puller/postgresPuller.js (hunk 1 above)
Comment (chenrozenes) [reply src/lib/puller/postgresPuller.js]: Agree.
We'll do it later when we'll need it

**Pull request merged**